### PR TITLE
Add documentation for gcp workload identity

### PR DESCRIPTION
**Pull request by akhil-rane**
The documentation includes
- overview of workload identity
- how do the new credentials look
- how to install openshift cluster with workload identity
- verification post installation
- how to do cleanup after uninstalling a cluster

x-ref: https://issues.redhat.com/browse/CCO-117

### CODE REVIEW

File: docs/workload_identity.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+   "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/test-service-account-42ssv@test-project.iam.gserviceaccount.com"
+}
+```
+
+With Workload Identity, the type of the credentials is `external_account`. `audience` is the target audience which is workload identity provider. `service_account_impersonation_url` key contains the resource url of the service account that can be impersonated with these credentials. `credentials_source.file` is the path to the oidc token, which is exchanged for google access token. The oidc token is rotated every one hour and thus credentials are short lived.
+
+```json
+{
+   "type": "external_account",
+   "audience": "//iam.googleapis.com/projects/725929187604/locations/global/workloadIdentityPools/test-pool/providers/test-provider",
```
[the remark this answers is not in the document]
Comment (akhil-rane) [reply docs/workload_identity.md:34]: replacing with dummy project number

File: docs/workload_identity.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+### Cleanup GCP resources after uninstalling the cluster
+
+Make sure you clean up the following resources after you uninstall your cluster. You can use `gcp_infra_name` used in installation step 7 to identify these resources
+
+1. Workload Identity Pool and Provider
+2. Cloud storage bucket used to store OpenID Connect configuration and the public key
+3. IAM service accounts created by ccoctl tool along with project policy bindings
+
+You can use `ccoctl gcp delete` command to delete all the above resources.
```
[the remark this answers is not in the document]
Comment (akhil-rane) [reply docs/workload_identity.md:128]: I just wanted to list out the command name here, not the entire command along with params. but that also works.

File: docs/gcp_workload_identity.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+   sample output of the above command
+   ```json
+   {
+      "type": "external_account",
+      "audience": "//iam.googleapis.com/projects/123456789/locations/global/workloadIdentityPools/test-pool/providers/test-provider",
+      "subject_token_type": "urn:ietf:params:oauth:token-type:jwt",
+      "token_url": "https://sts.googleapis.com/v1/token",
+      "service_account_impersonation_url": "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/test-openshift-image-registry-42ssv@test-project.iam.gserviceaccount.com:generateAccessToken",
+      "credential_source": {
+         "file": "var/run/secrets/openshift/serviceaccount/token",
```
Comment (joelddiaz) [inline docs/gcp_workload_identity.md:106]: whoops, you took my literal suggestion which had a typo :)
```suggestion
         "file": "/var/run/secrets/openshift/serviceaccount/token",
```

Comment (akhil-rane) [reply docs/gcp_workload_identity.md]: fixed

File: docs/gcp_workload_identity.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+   }
+}
+```
+
+### Steps to install an OpenShift Cluster with Workload Identity
+
+1. Set $RELEASE_IMAGE to point to a sufficiently new OpenShift release
+2. Extract the GCP Credentials Request objects from the above release image. You must use 4.7 or newer version of `oc` CLI.
+   ```
+   mkdir credreqs ; oc adm release extract --cloud=gcp --credentials-requests $RELEASE_IMAGE --to=./credreqs ; cat ./credreqs/*.yaml > credreqs.yaml
```
Comment (lwan-wanglin) [inline docs/gcp_workload_identity.md:52]: @akhil-rane For step 2, we don't need to make credreqs to one file, right?

File: docs/gcp_workload_identity.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+### Cleanup GCP resources after uninstalling the cluster
+
+Make sure you clean up the following resources after you uninstall your cluster. You can use `gcp_infra_name` used in installation step 7 to identify these resources
+
+1. Workload Identity Pool and Provider
+2. Cloud storage bucket used to store OpenID Connect configuration and the public key
+3. IAM service accounts created by ccoctl tool along with project policy bindings
+
+You can use `ccoctl gcp delete --name=<name> --region=<gcp-region>` command to delete all the above resources.
```
Comment (lwan-wanglin) [inline docs/gcp_workload_identity.md:122]: ccoctl gcp delete command needs "--project=<gcp-project-id>" parameter
`ccoctl gcp delete --name=<name> --region=<gcp-region> --project=<gcp-project-id>`

File: docs/gcp_workload_identity.md (hunk 4 above)
Comment (akhil-rane) [reply docs/gcp_workload_identity.md:52]: Nope, I will fix it. Nice catch!

File: docs/gcp_workload_identity.md (hunk 5 above)
Comment (akhil-rane) [reply docs/gcp_workload_identity.md:122]: agreed.

**Comment by akhil-rane** [comment]
@joelddiaz this is ready for a final review

**Review by joelddiaz — COMMENTED** [review]
/lgtm

**Comment by akhil-rane** [comment]
/label px-approved

the parent [epic](https://issues.redhat.com/browse/CCO-114) has been px-approved by @sferich888

**Comment by akhil-rane** [comment]
@lwan-wanglin addressed your comments. PTAL

File: docs/gcp_workload_identity.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+### Changes in the Credentials Secret with STS
```
Comment (jeana-redhat) [inline docs/gcp_workload_identity.md:9]: s/STS/Workload Identity ?

**Review by jeana-redhat — COMMENTED** [review]
This looks great! I left a couple comments but this is an excellent resource for me to use for official product docs. Thanks :)

File: docs/gcp_workload_identity.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
+# Short lived credentials with GCP Workload Identity Federation
+### Overview
+OpenShift can be configured to use temporary credentials for different components with GCP Workload Identity. It enables an authentication flow allowing a component to assume an IAM Service Account resulting in short-lived credentials. It also automates requesting and refreshing of credentials using a OIDC identity provider. OpenShift can sign ServiceAccount tokens trusted by the provider which can be projected into a Pod and used for authentication. The following is a diagram showing how it works.
+
+![workload_identity flow](gcp_workload_identity_flow.png)
```
[the remark this answers is not in the document]
Comment (akhil-rane) [reply docs/gcp_workload_identity.md:5]: sure that will great!

**Pull request merged**